File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.py
# Last updated: 7/9/2026, 10:15:17 AM
class Solution(object):
    def maxProfit(self, prices):
        min,max=prices[0],0
        for i in prices:
            if(min>i):
                min=i
            if i-min > max :
                max=i-min
        return max
        

File: 121-BestTimeToBuyAndSellStock/121-BestTimeToBuyAndSellStock.java
// Last updated: 7/9/2026, 10:15:10 AM
class Solution {
    public int maxProfit(int[] prices) {
        int l=prices[0];
        int r=1;
        int ans=0;
        while(r<prices.length){
            if(prices[r]>l){
                int profit=prices[r]-l;
                ans=Math.max(ans,profit);
            }else{
                l=prices[r];
            }
            r++;
        }
        return ans;
    }
}